lender.hpp

```diff
@@ -26,22 +26,12 @@ class recycle_memory;
 template <typename T>
 class buffer_ptr;
 
-template <typename T>
-class reuseable_buffer {
-    friend class buffer_ptr<T>;
-    friend class unit_test;
-    
-    private:
-        recycle_memory<T>& recycle;
-        // if this was shared_ptr, would memory be deleted at destruction?
-        T* ptr; 
-
-    public: 
-        std::vector<size_t> shape;
-
-        reuseable_buffer(T* p, recycle_memory<T>& r);
-        ~reuseable_buffer();
-        auto operator[](unsigned int i) const noexcept -> T&;
+struct buffer_deleter {
+    template <typename T>
+    static auto
+    return_to_recycler(recycle_memory<T>& recycle) {
+        return [&](T* ptr){ recycle.return_memory(ptr); };

```

Comment:
My understanding of this line is shaky, would you please explain the syntax?

Reply:
Also should buffer deleter be a proper functor? Otherwise, why have `return_to_recycler` in a new struct rather than using the `recycle_memory` function?